streams/src/test/java/org/apache/kafka/streams/processor/internals/StoreChangelogReaderTest.java

```diff
@@ -943,6 +946,31 @@ public void shouldThrowIfRestoreCallbackThrows() {
         assertEquals(kaboom, thrown.getCause());
     }
 
+    @Test
+    public void shouldThrowIfRestoreGetInstanceFenced() {
```

Comment:
The restore consumer should never hit this error. Hence, I am wondering if we actually need this test? Does it make sense to test for something that will/should never happen?

We should rather have a test that the restore consumer is never configured with an `group.instance.id` config if we don't have such a test.

Reply:
Actually we do not forbid users to set instance.id in restore.consumer, it's just that since restore consumer never uses `subscribe` it would not join-group and hence should not get this exception ever -- so personally I agree with you that this test is not very significant by itself but just to tighten the logic in case the restore consumer calls `subscribe` (mistakenly?) later.

Reply:
Should we improve `StreamsConfig#getRestoreConsumerConfigs()` (and also for global consumer) accordingly, to remove `group.instance.id` if set and log a INFO message? (Plus a corresponding test?)

We can of course still keep this test. But if we test that we don't catch `FencedInstanceIdException` on the restore consumer, should we add corresponding tests for the main and global consumer, too?

Reply:
I'm more inclined to @mjsax idea here, we should probably just ensure no instance.id being set for restore consumer in the config level, instead of doing a unit test.